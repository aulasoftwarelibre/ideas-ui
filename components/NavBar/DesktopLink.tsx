import Link from 'next/link';
import { NextRouter, withRouter } from 'next/router';
import React from 'react';

interface Props {
  label: string;
  path: string;
  router: NextRouter;
}

const DesktopLink: React.FunctionComponent<Props> = ({
  label,
  path,
  router,
}) => {
  const isActive: boolean = router.pathname === path;

  if (isActive) {
    return (
      <span className="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-ideaOrange">
        {label}
      </span>
    );
  }

  return (
    <Link href={path}>
      <a className="ml-4 px-3 py-2 text-sm font-medium leading-5 text-ideaOrange border-b-2 border-transparent hover:border-ideaOrange">
        {label}
      </a>
    </Link>
  );
};

export default withRouter(DesktopLink);
