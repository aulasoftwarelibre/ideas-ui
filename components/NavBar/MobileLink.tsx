import Link from 'next/link';
import { NextRouter, withRouter } from 'next/router';
import React from 'react';

export interface Props {
  label: string;
  path: string;
  router: NextRouter;
}

const MobileLink: React.FunctionComponent<Props> = ({
  label,
  path,
  router,
}) => {
  const isActive: boolean = router.pathname == path;

  if (isActive) {
    return (
      <span
        className="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        data-testid="active-mobile-link"
      >
        {label}
      </span>
    );
  }

  return (
    <Link href={path}>
      <a
        className="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        data-testid="mobile-link"
      >
        {label}
      </a>
    </Link>
  );
};

export default withRouter(MobileLink);
