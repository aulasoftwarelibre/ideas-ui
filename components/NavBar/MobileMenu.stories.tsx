import { Story } from '@storybook/react/types-6-0';
import React from 'react';
import { withNextRouter } from 'storybook-addon-next-router';

import MobileMenu, { Props } from './MobileMenu';

export default {
  component: MobileMenu,
  title: "Navbar/MobileMenu",
  decorators: [withNextRouter],
  parameters: {
    viewport: {
      defaultViewport: "mobile1",
    },
  },
};

const Template: Story<Props> = (args) => <MobileMenu {...args} />;

export const WithoutActiveLink = Template.bind({});
WithoutActiveLink.args = {
  hideMenu: false,
};
WithoutActiveLink.parameters = {
  nextRouter: {
    pathname: "/user/profile",
  },
};

export const WithActiveLink = Template.bind({});
WithActiveLink.args = {
  hideMenu: false,
};

export const Hidden = Template.bind({});
Hidden.args = {
  hideMenu: true,
};
