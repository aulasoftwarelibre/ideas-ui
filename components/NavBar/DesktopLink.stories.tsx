import React from 'react';
import { withNextRouter } from 'storybook-addon-next-router';

import DesktopLink from './DesktopLink';

export default {
  component: DesktopLink,
  title: "Navbar/DesktopLink",
  decorators: [withNextRouter],
};

export const Default = () => {
  return <DesktopLink path="/no-active" label="Groups" />;
};

Default.parameters = {
  nextRouter: {
    pathname: "/active",
  },
};

export const Active = () => {
  return <DesktopLink path="/active" label="Groups" />;
};

Active.parameters = {
  nextRouter: {
    pathname: "/active",
  },
};
