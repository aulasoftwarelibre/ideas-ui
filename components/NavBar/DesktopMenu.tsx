import React from "react";
import classnames from "classnames";
import GradientButton from "../GradientButton";
import Link from "next/link";
import DesktopLink from "./DesktopLink";

interface Props {
  hideMenu: boolean;
  hideProfile: boolean;
  closeMenuHandler: any;
  closeProfileHandler: any;
}

export const DesktopMenu: React.FunctionComponent<Props> = ({
  hideMenu,
  hideProfile,
  closeMenuHandler,
  closeProfileHandler,
}) => {
  return (
    <div className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8">
      <div className="relative flex items-center justify-between h-16">
        <div className="absolute inset-y-0 left-0 flex items-center sm:hidden">
          <button
            className="inline-flex items-center justify-center p-2 rounded-md text-ideaOrange hover:text-white hover:bg-ideaOrange focus:outline-none focus:bg-ideaOrange focus:text-white transition duration-150 ease-in-out"
            aria-label="Main menu"
            aria-expanded="false"
            onClick={closeMenuHandler}
          >
            <svg
              className={classnames(
                "h-6 w-6",
                { block: hideMenu },
                { hidden: !hideMenu }
              )}
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                d="M4 6h16M4 12h16M4 18h16"
              />
            </svg>
            <svg
              className={classnames(
                "h-6 w-6",
                { hidden: hideMenu },
                { block: !hideMenu }
              )}
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                d="M6 18L18 6M6 6l12 12"
              />
            </svg>
          </button>
        </div>
        <div className="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start">
          <div className="flex-shrink-0">
            <img
              className="block lg:hidden h-8 w-auto"
              src="/ideas-isotipo-naranja.svg"
              alt="Workflow logo"
            />
            <img
              className="hidden lg:block h-8 w-auto"
              src="/ideas-imagotipo-naranja.svg"
              alt="Workflow logo"
            />
          </div>
          <div className="hidden sm:block sm:ml-6">
            <div className="flex">
              <DesktopLink title="Inicio" path="/"></DesktopLink>
              <DesktopLink title="Organizaciones" path="/groups"></DesktopLink>
              <DesktopLink title="Ayuda" path="/help"></DesktopLink>
            </div>
          </div>
        </div>
        <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0">
          <div className="ml-3 relative">
            <div>
              <button
                className="flex text-sm border-2 border-transparent rounded-full focus:outline-none focus:border-white transition duration-150 ease-in-out"
                id="user-menu"
                aria-label="User menu"
                aria-haspopup="true"
                onClick={closeProfileHandler}
              >
                <img
                  className="h-8 w-8 rounded-full"
                  src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                  alt=""
                />
              </button>
            </div>
            <div className="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg">
              <div
                className={classnames("py-1 rounded-md bg-white shadow-xs", {
                  hidden: hideProfile,
                })}
                role="menu"
                aria-orientation="vertical"
                aria-labelledby="user-menu"
              >
                <a
                  href="#"
                  className="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  role="menuitem"
                >
                  Your Profile
                </a>
                <a
                  href="#"
                  className="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  role="menuitem"
                >
                  Settings
                </a>
                <a
                  href="#"
                  className="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  role="menuitem"
                >
                  Sign out
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DesktopMenu;
