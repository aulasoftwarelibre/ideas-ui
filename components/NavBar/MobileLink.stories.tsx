import React from 'react';
import { withNextRouter } from 'storybook-addon-next-router';

import MobileLink from './MobileLink';

export default {
  component: MobileLink,
  title: "Navbar/MobileLink",
  decorators: [withNextRouter],
};

export const Default = () => {
  return <MobileLink path="/no-active" label="Groups" />;
};

Default.story = {
  parameters: {
    nextRouter: {
      pathname: "/active",
    },
  },
};

export const Active = () => {
  return <MobileLink path="/active" label="Profile" />;
};

Active.story = {
  parameters: {
    nextRouter: {
      pathname: "/active",
    },
  },
};
