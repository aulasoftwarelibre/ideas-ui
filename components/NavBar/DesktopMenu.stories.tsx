import React from 'react';
import { withNextRouter } from 'storybook-addon-next-router';

import DesktopMenu from './DesktopMenu';

export default {
  component: DesktopMenu,
  title: "Navbar/DesktopMenu",
  decorators: [withNextRouter],
};

export const WithActiveLink = () => {
  return <DesktopMenu />;
};

export const WithoutActiveLink = () => {
  return <DesktopMenu />;
};

WithoutActiveLink.parameters = {
  nextRouter: {
    pathname: "/no-active",
  },
};
