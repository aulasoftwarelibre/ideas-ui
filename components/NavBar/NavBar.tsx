import React from "react";
import DesktopMenu from "./DesktopMenu";
import { MobileMenu } from "./MobileMenu";
import { useSession } from "next-auth/client";

import { Session } from "../../model/user/auth";
import GradientButton from "../GradientButton";
import UserMenu from "./UserMenu";
import Link from "next/link";

interface Props {
  session: Session;
}

export const NavBar: React.FunctionComponent<Props> = ({ session }) => {
  const [hideMenu, setMenu] = React.useState(true);
  const [hideProfile, setProfile] = React.useState(true);

  return (
    <nav className="bg-white">
      <div className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8">
        <div className="relative flex items-center justify-between h-16">
          <DesktopMenu
            hideMenu={hideMenu}
            closeMenuHandler={() => setMenu(!hideMenu)}
          ></DesktopMenu>
          {!session && (
            <Link href="/api/auth/signin">
              <GradientButton>Iniciar sesión</GradientButton>
            </Link>
          )}
          {session && (
            <>
              <UserMenu
                session={session}
                hideMenu={hideProfile}
                onClick={() => setProfile(!hideProfile)}
              />
              <GradientButton>Añadir idea</GradientButton>
            </>
          )}
        </div>
      </div>
      <MobileMenu
        hideMenu={hideMenu}
        closeMenuHandler={() => setMenu(!hideMenu)}
      ></MobileMenu>
    </nav>
  );
};

export default NavBar;
