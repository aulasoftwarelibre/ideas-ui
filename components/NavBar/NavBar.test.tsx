import React from 'react';

import { renderMockRouter, screen } from '../../utils/test-utils';
import { Props } from './NavBar';
import { DesktopWithoutSession, DesktopWithProfile, DesktopWithSession } from './NavBar.stories';

describe("NavBar", () => {
  it("should render desktop menu without session", () => {
    renderMockRouter(
      <DesktopWithoutSession {...(DesktopWithoutSession.args as Props)} />
    );

    expect(screen.getAllByText("Iniciar sesión"));
  });

  it("should render desktop menu with session", () => {
    renderMockRouter(
      <DesktopWithSession {...(DesktopWithSession.args as Props)} />
    );

    expect(screen.getAllByText("Añadir idea"));
  });

  it("should render desktop menu with profile menu open", () => {
    renderMockRouter(<DesktopWithProfile />);

    expect(screen.getByTestId("profile-menu"));
  });
});
