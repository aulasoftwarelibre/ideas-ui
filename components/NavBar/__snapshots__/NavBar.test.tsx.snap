// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render with mobile menu open 1`] = `
<DocumentFragment>
  <nav
    class="bg-white"
  >
    <div
      class="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
    >
      <div
        class="relative flex items-center justify-between h-16"
      >
        <div
          class="absolute inset-y-0 left-0 flex items-center sm:hidden"
        >
          <button
            aria-expanded="false"
            aria-label="Main menu"
            class="inline-flex items-center justify-center p-2 rounded-md text-primary hover:text-white hover:bg-primary focus:outline-none focus:bg-primary focus:text-white transition duration-150 ease-in-out"
          >
            <svg
              class="h-6 w-6 hidden"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="h-6 w-6 block"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start"
        >
          <div
            class="flex-shrink-0"
          >
            <img
              alt="Workflow logo"
              class="block lg:hidden h-8 w-auto"
              src="/ideas-isotipo-naranja.svg"
            />
            <img
              alt="Workflow logo"
              class="hidden lg:block h-8 w-auto"
              src="/ideas-imagotipo-naranja.svg"
            />
          </div>
          <div
            class="hidden sm:block sm:ml-6"
          >
            <div
              class="flex"
            >
              <span
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-primary"
              >
                Inicio
              </span>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/groups"
              >
                Grupos
              </a>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/help"
              >
                Ayuda
              </a>
            </div>
          </div>
        </div>
        <div
          class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
        >
          <div
            class="relative"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-label="User menu"
                class="flex text-sm border-2 border-transparent rounded-full focus:outline-none focus:border-white transition duration-150 ease-in-out"
                id="user-menu"
              >
                <img
                  alt="user"
                  class="h-8 w-8 rounded-full"
                  src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50?f=y"
                />
              </button>
            </div>
            <div
              class="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg"
            >
              <div
                aria-labelledby="user-menu"
                aria-orientation="vertical"
                class="py-1 rounded-md bg-white shadow-xs hidden"
                role="menu"
              >
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/user/profile"
                  role="menuitem"
                >
                  Editar perfil
                </a>
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/user/ideas"
                  role="menuitem"
                >
                  Actividades
                </a>
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/api/auth/signout"
                  role="menuitem"
                >
                  Salir
                </a>
              </div>
            </div>
          </div>
        </div>
        <span
          class="hidden sm:block"
        >
          <a
            href="/idea/add"
          >
            <button
              class="ml-3  bg-gradient-l-primary-to-light hover:bg-none hover:bg-secondary text-white font-bold py-2 px-6 text-sm rounded-md"
            >
              Añadir idea
            </button>
          </a>
        </span>
      </div>
    </div>
    <div
      class="sm:hidden block"
    >
      <div
        class="px-2 pt-2 pb-3 bg-white"
      >
        <span
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        >
          Inicio
        </span>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/groups"
        >
          Grupos
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/ideas/new"
        >
          Añadir idea
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/help"
        >
          Ayuda
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar should render with profile menu open 1`] = `
<DocumentFragment>
  <nav
    class="bg-white"
  >
    <div
      class="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
    >
      <div
        class="relative flex items-center justify-between h-16"
      >
        <div
          class="absolute inset-y-0 left-0 flex items-center sm:hidden"
        >
          <button
            aria-expanded="false"
            aria-label="Main menu"
            class="inline-flex items-center justify-center p-2 rounded-md text-primary hover:text-white hover:bg-primary focus:outline-none focus:bg-primary focus:text-white transition duration-150 ease-in-out"
          >
            <svg
              class="h-6 w-6 block"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="h-6 w-6 hidden"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start"
        >
          <div
            class="flex-shrink-0"
          >
            <img
              alt="Workflow logo"
              class="block lg:hidden h-8 w-auto"
              src="/ideas-isotipo-naranja.svg"
            />
            <img
              alt="Workflow logo"
              class="hidden lg:block h-8 w-auto"
              src="/ideas-imagotipo-naranja.svg"
            />
          </div>
          <div
            class="hidden sm:block sm:ml-6"
          >
            <div
              class="flex"
            >
              <span
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-primary"
              >
                Inicio
              </span>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/groups"
              >
                Grupos
              </a>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/help"
              >
                Ayuda
              </a>
            </div>
          </div>
        </div>
        <div
          class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
        >
          <div
            class="relative"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-label="User menu"
                class="flex text-sm border-2 border-transparent rounded-full focus:outline-none focus:border-white transition duration-150 ease-in-out"
                id="user-menu"
              >
                <img
                  alt="user"
                  class="h-8 w-8 rounded-full"
                  src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50?f=y"
                />
              </button>
            </div>
            <div
              class="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg"
            >
              <div
                aria-labelledby="user-menu"
                aria-orientation="vertical"
                class="py-1 rounded-md bg-white shadow-xs"
                role="menu"
              >
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/user/profile"
                  role="menuitem"
                >
                  Editar perfil
                </a>
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/user/ideas"
                  role="menuitem"
                >
                  Actividades
                </a>
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/api/auth/signout"
                  role="menuitem"
                >
                  Salir
                </a>
              </div>
            </div>
          </div>
        </div>
        <span
          class="hidden sm:block"
        >
          <a
            href="/idea/add"
          >
            <button
              class="ml-3  bg-gradient-l-primary-to-light hover:bg-none hover:bg-secondary text-white font-bold py-2 px-6 text-sm rounded-md"
            >
              Añadir idea
            </button>
          </a>
        </span>
      </div>
    </div>
    <div
      class="sm:hidden hidden"
    >
      <div
        class="px-2 pt-2 pb-3 bg-white"
      >
        <span
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        >
          Inicio
        </span>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/groups"
        >
          Grupos
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/ideas/new"
        >
          Añadir idea
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/help"
        >
          Ayuda
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar should render with session 1`] = `
<DocumentFragment>
  <nav
    class="bg-white"
  >
    <div
      class="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
    >
      <div
        class="relative flex items-center justify-between h-16"
      >
        <div
          class="absolute inset-y-0 left-0 flex items-center sm:hidden"
        >
          <button
            aria-expanded="false"
            aria-label="Main menu"
            class="inline-flex items-center justify-center p-2 rounded-md text-primary hover:text-white hover:bg-primary focus:outline-none focus:bg-primary focus:text-white transition duration-150 ease-in-out"
          >
            <svg
              class="h-6 w-6 block"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="h-6 w-6 hidden"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start"
        >
          <div
            class="flex-shrink-0"
          >
            <img
              alt="Workflow logo"
              class="block lg:hidden h-8 w-auto"
              src="/ideas-isotipo-naranja.svg"
            />
            <img
              alt="Workflow logo"
              class="hidden lg:block h-8 w-auto"
              src="/ideas-imagotipo-naranja.svg"
            />
          </div>
          <div
            class="hidden sm:block sm:ml-6"
          >
            <div
              class="flex"
            >
              <span
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-primary"
              >
                Inicio
              </span>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/groups"
              >
                Grupos
              </a>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/help"
              >
                Ayuda
              </a>
            </div>
          </div>
        </div>
        <div
          class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
        >
          <div
            class="relative"
          >
            <div>
              <button
                aria-haspopup="true"
                aria-label="User menu"
                class="flex text-sm border-2 border-transparent rounded-full focus:outline-none focus:border-white transition duration-150 ease-in-out"
                id="user-menu"
              >
                <img
                  alt="user"
                  class="h-8 w-8 rounded-full"
                  src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50?f=y"
                />
              </button>
            </div>
            <div
              class="origin-top-right absolute right-0 mt-2 w-48 rounded-md shadow-lg"
            >
              <div
                aria-labelledby="user-menu"
                aria-orientation="vertical"
                class="py-1 rounded-md bg-white shadow-xs hidden"
                role="menu"
              >
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/user/profile"
                  role="menuitem"
                >
                  Editar perfil
                </a>
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/user/ideas"
                  role="menuitem"
                >
                  Actividades
                </a>
                <a
                  class="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 transition duration-150 ease-in-out"
                  href="/api/auth/signout"
                  role="menuitem"
                >
                  Salir
                </a>
              </div>
            </div>
          </div>
        </div>
        <span
          class="hidden sm:block"
        >
          <a
            href="/idea/add"
          >
            <button
              class="ml-3  bg-gradient-l-primary-to-light hover:bg-none hover:bg-secondary text-white font-bold py-2 px-6 text-sm rounded-md"
            >
              Añadir idea
            </button>
          </a>
        </span>
      </div>
    </div>
    <div
      class="sm:hidden hidden"
    >
      <div
        class="px-2 pt-2 pb-3 bg-white"
      >
        <span
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        >
          Inicio
        </span>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/groups"
        >
          Grupos
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/ideas/new"
        >
          Añadir idea
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/help"
        >
          Ayuda
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`NavBar should render without session 1`] = `
<DocumentFragment>
  <nav
    class="bg-white"
  >
    <div
      class="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
    >
      <div
        class="relative flex items-center justify-between h-16"
      >
        <div
          class="absolute inset-y-0 left-0 flex items-center sm:hidden"
        >
          <button
            aria-expanded="false"
            aria-label="Main menu"
            class="inline-flex items-center justify-center p-2 rounded-md text-primary hover:text-white hover:bg-primary focus:outline-none focus:bg-primary focus:text-white transition duration-150 ease-in-out"
          >
            <svg
              class="h-6 w-6 block"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="h-6 w-6 hidden"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start"
        >
          <div
            class="flex-shrink-0"
          >
            <img
              alt="Workflow logo"
              class="block lg:hidden h-8 w-auto"
              src="/ideas-isotipo-naranja.svg"
            />
            <img
              alt="Workflow logo"
              class="hidden lg:block h-8 w-auto"
              src="/ideas-imagotipo-naranja.svg"
            />
          </div>
          <div
            class="hidden sm:block sm:ml-6"
          >
            <div
              class="flex"
            >
              <span
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-primary"
              >
                Inicio
              </span>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/groups"
              >
                Grupos
              </a>
              <a
                class="ml-4 px-3 py-2 text-sm font-medium leading-5 text-primary border-b-2 border-transparent hover:border-primary"
                href="/help"
              >
                Ayuda
              </a>
            </div>
          </div>
        </div>
        <a
          href="/api/auth/signin"
        >
          <button
            class="ml-3  bg-gradient-l-primary-to-light hover:bg-none hover:bg-secondary text-white font-bold py-2 px-6 text-sm rounded-md"
          >
            Iniciar sesión
          </button>
        </a>
      </div>
    </div>
    <div
      class="sm:hidden hidden"
    >
      <div
        class="px-2 pt-2 pb-3 bg-white"
      >
        <span
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-white bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
        >
          Inicio
        </span>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/groups"
        >
          Grupos
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/ideas/new"
        >
          Añadir idea
        </a>
        <a
          class="mt-1 block px-3 py-2 rounded-md text-base font-medium text-primary hover:text-white hover:bg-primary focus:outline-none focus:text-white focus:bg-primary transition duration-150 ease-in-out"
          href="/help"
        >
          Ayuda
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
